File: frontend/src/features/alerts/types/alert.ts
export interface Alert {
  id: number;

  attack: string;

  severity: string;

  source_ip: string;

  destination_ip: string;

  description: string;

  timestamp: string;

  details: Record<string, unknown>;
}

export interface AlertsResponse {
  page: number;
  limit: number;
  total: number;
  pages: number;
  items: Alert[];
}

File: backend/app/schemas/alert.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AlertResponse(BaseModel):
    id: int
    attack: str
    severity: str
    source_ip: str
    destination_ip: str | None
    description: str
    details: dict[str, Any]
    timestamp: datetime

    model_config = ConfigDict(from_attributes=True)


class PaginatedAlertsResponse(BaseModel):
    page: int
    limit: int
    total: int
    pages: int
    items: list[AlertResponse]